File: class_tool/src/main/java/com/tempsfteam/class_tool/vo/SignInInfo.java
package com.tempsfteam.class_tool.vo;

import com.tempsfteam.class_tool.constant.role.IdentityEnum;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;

/**
 * @Description:
 * @Author: Zhan
 * @DateTime: 2023/7/30 14:08
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SignInInfo {
    /**
     * token
     */
    private String token;
    /**
     * 用户身份
     */
    private String userRole;
    /**
     * 用户权限列表
     */
    private Map<String, List<String>> permissionList;

    public void setUserRole(Integer userRole) {
        IdentityEnum identityEnum = IdentityEnum.fromValue(userRole);
        this.userRole = identityEnum.getUserRoleString();
    }
}

File: class_tool/src/main/resources/static/src/models/vo/signInInfo.js
export default {
    name: 'signInInfo',
    title: 'SignInInfo',
    fields: [
        { name: 'token', label: 'token', type: 'string' },
        { name: 'userRole', label: '用户角色字符串', type: 'string' },
        { name: 'permissionList', label: '权限列表 (map)', type: 'object' }
    ]
};